### PR TITLE
Validate capabilities on CRI-O start

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now validate the provided capabilities on config validation to give
users more feedback about their input.
#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
- Added verification of provided `--default-capabilities`/`default_capabilities` on configuration validation (CRI-O startup)
```

### CODE REVIEW

**Comment by haircommander** [comment]
LGTM

**Comment by rhatdan** [comment]
Is there any chance we could share code with containers/common/pkg/capabilities?
Potentially common/pkg/config?

**Comment by saschagrunert** [comment]
> Is there any chance we could share code with containers/common/pkg/capabilities?
> Potentially common/pkg/config?

Ah yeah I will merge those.

/hold

File: internal/config/capabilities/capabilities.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		"KILL",
+	}
+}
+
+// Validate checks if the provided capabilities are available on the system
+func (c Capabilities) Validate() error {
+	caps := Capabilities{}
+	for _, cap := range c {
+		caps = append(caps, "CAP_"+strings.ToUpper(cap))
+	}
```
Comment (saschagrunert) [inline internal/config/capabilities/capabilities.go]: We still have to pre-process them to maintain the original behavior.

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch

**Comment by saschagrunert** [comment]
@haircommander I think this might be ready to be merged.

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch

**Comment by TomSweeneyRedHat** [comment]
LGTM once tests are happy

**Comment by TomSweeneyRedHat** [comment]
@saschagrunert I think you might be able to rebase this now to get the lint failure to pass.

**Comment by saschagrunert** [comment]
Rebased on top of the latest master.

**Comment by haircommander** [comment]
@saschagrunert needs rebase

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch

**Pull request merged**